yellowbrick/text/dispersion.py

```diff
@@ -95,34 +120,91 @@ def fit(self, text):
         text : list
             Should be provided as a list of documents that contain
             a list of words in the order they appear in the document.
+
+        y : ndarray or Series of length n
+            An optional array or series of target or class values for
+            instances. If this is specified, then the points will be colored
+            according to their class.
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
         """
 
+        if y is not None:
+            self.classes_ = np.unique(y)
+        elif y is None and self.labels is not None:
+            self.classes_ = np.array([self.labels[0]])
+        else:
+            self.classes_ = np.array([self.NULL_CLASS])
+
         # Create an index (e.g. the y position) for the target words
         self.target_words_ = np.flip(self.words, axis=0)
```

Comment:
What happens if one or more of the `self.target_words_` isn't in the corpus?

Reply:
@rebeccabilbro Great Catch!  I came across this the other day and forgot about it.  If no target words are in the corpus then an error is raised.  

I now have to decide if I present a blank graph or raise a yellowbrickValueError in this case

Reply:
As a user, it would be great to get a custom `YellowbrickValueError` in this case! Maybe some thing like "Could not find {} in the corpus."

Reply:
I finally got a fix for this..it is in the latest commit